dubbo-common/src/main/java/org/apache/dubbo/config/AbstractInterfaceConfig.java

```diff
@@ -221,7 +221,7 @@
     /**
      * The url of the reference service
      */
-    protected final transient List<URL> urls = new ArrayList<URL>();
+    protected final transient List<URL> urls = new ArrayList<>();
```

Comment:
**Scalability Issue**: Using raw types for the ArrayList instantiation is not type-safe and may lead to ClassCastException at runtime when incorrect types are inserted into the list. This can potentially degrade the scalability of the system by introducing runtime errors that are hard to debug and fix. <br> **Fix**: Use generics for the ArrayList instantiation to ensure type safety. This will help the system scale better by preventing ClassCastException at runtime and making the code easier to maintain. <br> **Code Suggestion**: 
 ```
 -    protected final transient List<URL> urls = new ArrayList<URL>();
 +    protected final transient List<URL> urls = new ArrayList<>();

 Use generics for the ArrayList instantiation to ensure type safety. This will help the system scale better by preventing ClassCastException at runtime and making the code easier to maintain.
 ```